### PR TITLE
allow non-finalized Hom

### PR DESCRIPTION
**Pull request by heiderich**

### CODE REVIEW

File: gap/FunctorCategories.gi [hunk 1]
```diff
@@ -708,11 +708,14 @@ InstallMethodWithCache( Hom,
...
     
-    Finalize( Hom );
+    AddToToDoList( ToDoListEntry( [ [ Hom, "IsFinalized", true ] ], function() IdentityFunctor( Hom )!.UnderlyingFunctor := IdentityFunctor( C ); end ) );
     
-    IdentityFunctor( Hom )!.UnderlyingFunctor := IdentityFunctor( C );
-    
-    return Hom;
+    if ValueOption( "FinalizeCategory" ) = false then
+        return Hom;
+    else
```
[the remark this answers is not in the document]
Comment (heiderich) [reply gap/FunctorCategories.gi]: I know. Does it hurt?

File: gap/FunctorCategories.gi [hunk 2]
```diff
@@ -708,11 +708,14 @@ InstallMethodWithCache( Hom,
           
     fi;
     
-    Finalize( Hom );
+    AddToToDoList( ToDoListEntry( [ [ Hom, "IsFinalized", true ] ], function() IdentityFunctor( Hom )!.UnderlyingFunctor := IdentityFunctor( C ); end ) );
```
[the remark this answers is not in the document]
Comment (heiderich) [reply gap/FunctorCategories.gi]: Before the constructor contained the line

`IdentityFunctor( Hom )!.UnderlyingFunctor := IdentityFunctor( C );`

I wanted to keep the current behavior as closely as possible. So I merely delegated this to the ToDo system.

I also asked myself what the idea behind the original line was. It is due to you, so I'd like to give back this question to you.

File: gap/FunctorCategories.gi (hunk 1 above)
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply gap/FunctorCategories.gi]: It does not comply to the (unwritten) coding style of all my packages. I expect the generic return statement outside of any `if-elif-else` field.

[the remark this answers is not in the document]
Comment (heiderich) [reply gap/FunctorCategories.gi]: I see. I removed it.

File: gap/FunctorCategories.gi (hunk 2 above)
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply gap/FunctorCategories.gi]: Probably because it is used somewhere. If not the it is remnants of code from another package.

**Pull request merged**